pkg/querier/frontend/frontend.go

```diff
@@ -164,34 +164,46 @@ func (f *Frontend) handle(w http.ResponseWriter, r *http.Request) {
 	resp, err := f.roundTripper.RoundTrip(r)
 	queryResponseTime := time.Since(startTime)
 
-	if f.cfg.LogQueriesLongerThan > 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
+	if err != nil {
+		writeError(w, err)
+	} else {
+		hs := w.Header()
+		for h, vs := range resp.Header {
+			hs[h] = vs
+		}
+		w.WriteHeader(resp.StatusCode)
+		io.Copy(w, resp.Body)
+	}
+
+	// If LogQueriesLongerThan is set to <0 we log every query
+	if f.cfg.LogQueriesLongerThan != 0 && queryResponseTime > f.cfg.LogQueriesLongerThan {
 		logMessage := []interface{}{
-			"msg", "slow query",
+			"msg", "slow query detected",
+			"method", r.Method,
 			"host", r.Host,
 			"path", r.URL.Path,
 			"time_taken", queryResponseTime.String(),
 		}
-		for k, v := range r.URL.Query() {
-			logMessage = append(logMessage, fmt.Sprintf("qs_%s", k), strings.Join(v, ","))
+
+		// Extract the ID from the header of the request
+		id, _, err := user.ExtractOrgIDFromHTTPRequest(r)
+		if err != nil {
+			logMessage = append(logMessage, "tenant_id", id)
```

Comment:
Logging with `util.WithContext()` (done below) already injects the `org_id`. Why do we need to add it here too?

Reply:
It does, sorry I was confused since the weaveworks common server logs do not include the org_id